client/src/components/CheckBox/CheckBox.stories.tsx

```diff
@@ -8,7 +8,13 @@ export default {
   component: CheckBox,
 } as Meta;
 
-type CheckBoxProps = React.ComponentProps<typeof CheckBox>;
+type ComponentProps<T> = T extends
+  | React.ComponentType<infer P>
+  | React.Component<infer P>
+  ? JSX.LibraryManagedAttributes<T, P>
+  : never;
```

Comment:
Since we'll want to be reusing this a bunch, and may or may not have to update the pattern in the future, please move it to `src/types/utility_types.d.ts` and import from there as needed :+1: